### PR TITLE
guest_os_booting: add new case for ovmf nvram

### PR DESCRIPTION
**Pull request by meinaLi**
Automate case:
VIRT-297064 - [OVMF] Boot vm with ovmf nvram

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --vt-machine-type q35 --test-runner=runner guest_os_booting.ovmf_nvram
JOB ID     : c318ba6edcee0875b55ae03677c7e57ecd37483d
JOB LOG    : /var/lib/avocado/job-results/job-2023-06-09T03.24-c318ba6/job.log
 (1/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_nvram.positive_test.template_enable_secure: PASS (79.83 s)
 (2/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_nvram.positive_test.template_disable_secure: PASS (84.38 s)
 (3/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_nvram.positive_test.manual_nvram_file: PASS (85.90 s)
 (4/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_nvram.negative_test.nonexist_template: PASS (19.44 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-06-09T03.24-c318ba6/results.html
JOB TIME   : 271.28 s
```

**Comment by meinaLi** [comment]
Label: modular_redesign

**Review by nanli1 — APPROVED** [review]

**Review by cliping — APPROVED** [review]
LGTM

File: libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_nvram.cfg [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            variants:
+                - template_enable_secure:
+                    template_path = "/usr/share/edk2/ovmf/OVMF_VARS.secboot.fd"
+                    nvram_dict = {'secure': 'yes', 'loader_readonly': 'yes', 'loader_type': 'pflash', 'loader': '${loader_path}', 'nvram_attrs': {'template': '${template_path}'}}
+                - template_disable_secure:
+                    template_path = "/usr/share/edk2/ovmf/OVMF_VARS.fd"
+                    nvram_dict = {'secure': 'yes', 'loader_readonly': 'yes', 'loader_type': 'pflash', 'loader': '${loader_path}', 'nvram_attrs': {'template': '${template_path}'}}
+                - manual_nvram_file:
+                    nvram_file = "/tmp/nvram"
+                    nvram_dict = {'secure': 'yes', 'loader_readonly': 'yes', 'loader_type': 'pflash', 'loader': '${loader_path}', 'nvram': '${nvram_file}'}
```
Comment (dzhengfy) [inline libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_nvram.cfg:18]: As all 4 cases have this same parameter and values are same, except this one. I suggest we move this parameter to above line 7 and keep this parameter here for this case. That will be cleaner.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_nvram.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+#   Author: Meina Li <meili@redhat.com>
+#
+from virttest.libvirt_xml import vm_xml
+from provider.guest_os_booting import guest_os_booting_base as guest_os
+
+
+def run(test, params, env):
+    """
+    This case it to very the ovmf nvram elements.
```
Comment (dzhengfy) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_nvram.py:16]: It seems this sentence need refine.

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_nvram.cfg (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_nvram.cfg:17]: I've merged this parameter to line 7. To match this, I added a conditional judgment in python file for template_path. Please review it again.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_nvram.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_nvram.py]: Updated.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**